src/material/datepicker/calendar-body.ts

```diff
@@ -356,6 +364,26 @@ export class MatCalendarBody implements OnChanges, OnDestroy, AfterViewChecked {
     return isInRange(value, this.previewStart, this.previewEnd, this.isRange);
   }
 
+  /** Gets ids of aria descriptions for the start and end of a date range. */
+  _getDescribedby(value: number): string | null {
+    if (!this.isRange) {
+      return null;
+    }
+    const ids: string[] = [];
```

Comment:
This is going to be called on each change detection which will generate a bunch of unnecessary arrays. Maybe we can only compute this when the IDs are assigned?

Reply:
good catch, I refactored this method to not construct any arrays.